akka-http2-support/src/main/scala/akka/http/scaladsl/model/http2/RequestResponseAssociation.scala

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.model.http2
+
+/**
+ * A marker trait for attribute values that should be (automatically) carried over from request to response.
+ */
+trait RequestResponseAssociation
```

Comment:
We might want to move to akka-http-core and make use of it there. We could use it also for the regular pools to associate requests with responses (currently, we create a `RequestContext` there to carry over a Promise from request to response which we could get rid of by using an (internal) attribute instead).